ui: cosmetic upgrade for Logical Plans

celiala opened the pull request:
This commit has cosmetic updates for Logical Plans, based on customer feedback.

Before:
<img width="400" alt="before" src="https://user-images.githubusercontent.com/3051672/54839778-f8418580-4ca1-11e9-8b21-9c5535a6b288.gif">

After:
<img width="400" alt="after" src="https://user-images.githubusercontent.com/3051672/54855140-91d15d00-4ccb-11e9-933a-fef3e7549e25.gif">

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/36062)
<!-- Reviewable:end -->

celiala commented:
Feedback from Jeremy/Piyush is that plans can potentially be long and take up too much vertical space. Solution is to set initial max height to 400px. Then user can click a "see full plan" to remove max-height and view full plan.

celiala committed:
ui: cosmetic upgrade for Logical Plans

This commit has cosmetic updates for Logical Plans, based
on customer feedback.

Before:
<img width="400" alt="before" src="https://user-images.githubusercontent.com/3051672/54839778-f8418580-4ca1-11e9-8b21-9c5535a6b288.gif">

After:
<img width="400" alt="after" src="https://user-images.githubusercontent.com/3051672/54855140-91d15d00-4ccb-11e9-933a-fef3e7549e25.gif">

Release note: None

celiala commented:
This is ready for review:

- A collapse/expand toggle is displayed if plans are longer than 400px
- Unit tests have been added

vilterp commented on pkg/ui/src/views/statements/planView.tsx:
```diff
@@ -304,51 +273,96 @@ function PlanNodes(props: {
+
+  showExpandDirections() {
+    // Only show directions to show/hide the full plan if content is longer than its max-height.
+    const innerContainer = document.getElementById("plan-view-inner-container");
```
Using `document.getElementByID` to grab DOM elements from inside of a React component is discouraged; you're supposed to use refs instead. (i.e. this will break if we ever have two of them on the page, but I guess that's… unlikely). See this guide: https://www.javascriptstuff.com/use-refs-not-ids/

I realize we're under time pressure though. So… idk 😛

vilterp reviewed (approved):
Looks good, sorry for taking so long to get to this.

LGTM, with the one issue about refs.

vilterp commented:
Another idea to manage long plans is to allow you to collapse a node, including its children (I believe in the prev design you could only collapse attributes). This would allow you to collapse a big plan down to its high level structure, then explore the subtree you're interested in.

celiala commented:
TFTR, Pete!

bors r+

> Best practice is to use refs instead of id's

Ah, that makes total sense - updated to use refs

> Another idea is to collapse a node, including its children...

Ooh, nice -- did a quick mock of this to feedback from Piyush/Jeremy (#36291):

Initial thoughts were that we'd want more design if we wanted to move forward with this [e.g. having the arrow on the right implies it should collapse just the attributes: we may want to move it to the left/replace the left-side bullet and perhaps have the gray node boxes encapsulate the children nodes].
